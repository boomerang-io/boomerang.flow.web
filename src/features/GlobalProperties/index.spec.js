import React from "react";
import { GlobalPropertiesContainer } from ".";
import { render, fireEvent, waitFor } from "@testing-library/react";

const mockfn = jest.fn();

const props = {
  globalConfiguration: {
    isFetching: false,
    status: "success",
    data: [
      {
        value: "12345",
        id: "5cdc5ad7460edb4a230b579a",
        key: "test.rtl",
        label: "Test RTL",
        type: "text"
      }
    ]
  },
  actions: {
    reset: mockfn,
    fetch: mockfn,
    addPropertyInStore: mockfn,
    deletePropertyInStore: mockfn,
    updatePropertyInStore: mockfn
  }
};

beforeEach(() => {
  document.body.setAttribute("id", "app");
});

describe("GlobalPropertiesContainer --- Snapshot Test", () => {
  it("Capturing Snapshot of GlobalPropertiesContainer", async() => {
    const { baseElement } = render(<GlobalPropertiesContainer {...props} />);
    expect(baseElement).toMatchSnapshot();
    await waitFor(() => {});
  });
});

describe("GlobalPropertiesContainer --- RTL", () => {
  it("Render the table and search correctly", async () => {
    const { getByText, getByPlaceholderText, queryAllByText } = render(<GlobalPropertiesContainer {...props} />);
    getByText(/Set global properties that are available for all Workflows/i);
    expect(queryAllByText(/Test RTL/i).length).toBe(1);
    const searchProperty = getByPlaceholderText(/Search/i);
    fireEvent.change(searchProperty, { target: { value: "2" } });
    expect(queryAllByText(/Test RTL/i).length).toBe(0);
    fireEvent.change(searchProperty, { target: { value: "Test" } });
    expect(queryAllByText(/Test RTL/i).length).toBe(1);
  });

  it("Opens create property modal", async () => {
    const { getByTestId, queryByText } = render(<GlobalPropertiesContainer {...props} />);
    expect(queryByText(/CREATE PROPERTY/)).not.toBeInTheDocument();

    const modalTrigger = getByTestId("create-global-configurations-property-button");
    fireEvent.click(modalTrigger);

    expect(queryByText(/Cancel/)).toBeInTheDocument();
  });

  it("Opens edit property modal", async () => {
    const { findByText, queryByText, getByTestId } = render(<GlobalPropertiesContainer {...props} />);

    expect(queryByText(/EDIT TEST RTL/i)).not.toBeInTheDocument();

    fireEvent.mouseOver(getByTestId("configuration-property-table-row"));

    const actionsMenu = getByTestId("configuration-property-table-overflow-menu");
    fireEvent.click(actionsMenu);

    const modalTrigger = await findByText(/Edit/i);
    fireEvent.click(modalTrigger);

    expect(queryByText(/EDIT TEST RTL/i)).toBeInTheDocument();
  });

  it("Opens delete property modal", async () => {
    const { findByText, queryByText, getByTestId } = render(<GlobalPropertiesContainer {...props} />);

    expect(queryByText(/DELETE TEST RTL/i)).not.toBeInTheDocument();

    fireEvent.mouseOver(getByTestId("configuration-property-table-row"));

    const actionsMenu = getByTestId("configuration-property-table-overflow-menu");
    fireEvent.click(actionsMenu);

    const modalTrigger = await findByText(/Delete/i);
    fireEvent.click(modalTrigger);

    expect(queryByText(/DELETE TEST RTL/i)).toBeInTheDocument();
  });
});
