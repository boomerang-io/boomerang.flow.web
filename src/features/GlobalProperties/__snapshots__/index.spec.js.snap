// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalPropertiesContainer --- Snapshot Test Capturing Snapshot of GlobalPropertiesContainer 1`] = `
<body
  id="app"
>
  <div>
    <div
      class="container"
    >
      <div
        class="container"
      >
        <div
          class="content"
        >
          <div
            class="info"
          >
            <h1
              class="title"
            >
              Properties
            </h1>
            <p
              class="description"
            >
              Set global properties that are accessible in flow processes.
            </p>
          </div>
          <section
            class="buttons"
          />
        </div>
      </div>
      <div
        class="tableContainer"
      >
        <div
          class="header"
        >
          <div
            class="bx--search bx--search--xl search"
          >
            <svg
              aria-hidden="true"
              class="bx--search-magnifier"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              style="will-change: transform;"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15 14.3L10.7 10c1.9-2.3 1.6-5.8-.7-7.7S4.2.7 2.3 3 .7 8.8 3 10.7c2 1.7 5 1.7 7 0l4.3 4.3.7-.7zM2 6.5C2 4 4 2 6.5 2S11 4 11 6.5 9 11 6.5 11 2 9 2 6.5z"
              />
            </svg>
            <label
              class="bx--label"
              for="properties-table-search"
            >
              Search
            </label>
            <input
              class="bx--search-input"
              id="properties-table-search"
              placeholder="Search"
              type="text"
            />
            <button
              aria-label="Clear search input"
              class="bx--search-close bx--search-close--hidden"
              type="button"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                height="20"
                preserveAspectRatio="xMidYMid meet"
                style="will-change: transform;"
                viewBox="0 0 32 32"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4l6.6 6.6L8 22.6 9.4 24l6.6-6.6 6.6 6.6 1.4-1.4-6.6-6.6L24 9.4z"
                />
              </svg>
            </button>
          </div>
          <button
            class="bx--btn bx--btn--field bx--btn--primary"
            data-testid="create-global-configurations-property-button"
            style="width: 12rem;"
            tabindex="0"
            type="button"
          >
            Create Property
            <svg
              aria-hidden="true"
              aria-label="Create Property"
              class="bx--btn__icon"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              style="will-change: transform;"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 7V3H7v4H3v2h4v4h2V9h4V7z"
              />
            </svg>
          </button>
        </div>
        <div
          class="bx--data-table-container"
        >
          <table
            class="bx--data-table bx--data-table--no-border"
          >
            <thead>
              <tr>
                <th
                  class="tableHeadHeader label"
                  id="label"
                  scope="col"
                >
                  <span
                    class="bx--table-header-label"
                  >
                    Label
                  </span>
                </th>
                <th
                  class="tableHeadHeader key"
                  id="key"
                  scope="col"
                >
                  <span
                    class="bx--table-header-label"
                  >
                    Key
                  </span>
                </th>
                <th
                  class="tableHeadHeader description"
                  id="description"
                  scope="col"
                >
                  <span
                    class="bx--table-header-label"
                  >
                    Description
                  </span>
                </th>
                <th
                  class="tableHeadHeader value"
                  id="value"
                  scope="col"
                >
                  <span
                    class="bx--table-header-label"
                  >
                    Value
                  </span>
                </th>
                <th
                  class="tableHeadHeader secured"
                  id="secured"
                  scope="col"
                >
                  <span
                    class="bx--table-header-label"
                  >
                    Secured
                  </span>
                </th>
                <th
                  class="tableHeadHeader actions "
                  id="actions "
                  scope="col"
                >
                  <span
                    class="bx--table-header-label"
                  >
                    Actions
                  </span>
                </th>
              </tr>
            </thead>
            <tbody
              aria-live="polite"
              class="tableBody"
            >
              <tr
                data-testid="configuration-property-table-row"
              >
                <td
                  style="padding: 0px;"
                >
                  <div
                    class="tableCell"
                  >
                    <p
                      class="tableTextarea"
                    >
                      Test RTL
                    </p>
                  </div>
                </td>
                <td
                  style="padding: 0px;"
                >
                  <div
                    class="tableCell"
                  >
                    <p
                      class="tableTextarea"
                    >
                      test.rtl
                    </p>
                  </div>
                </td>
                <td
                  style="padding: 0px;"
                >
                  <div
                    class="tableCell"
                  >
                    <p
                      class="tableTextarea"
                    >
                      ---
                    </p>
                  </div>
                </td>
                <td
                  style="padding: 0px;"
                >
                  <div
                    class="tableCell"
                  >
                    <p
                      class="tableTextarea"
                    >
                      12345
                    </p>
                  </div>
                </td>
                <td
                  style="padding: 0px;"
                >
                  <div
                    class="tableCell"
                  >
                    <svg
                      alt="unsecured"
                      aria-hidden="true"
                      class="tableSecured unsecured"
                      focusable="false"
                      height="32"
                      preserveAspectRatio="xMidYMid meet"
                      style="will-change: transform;"
                      viewBox="0 0 32 32"
                      width="32"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4l6.6 6.6L8 22.6 9.4 24l6.6-6.6 6.6 6.6 1.4-1.4-6.6-6.6L24 9.4z"
                      />
                    </svg>
                  </div>
                </td>
                <td
                  style="padding: 0px;"
                >
                  <div
                    class="tableCell"
                  >
                    <div
                      aria-expanded="false"
                      aria-haspopup="true"
                      aria-label="Overflow menu"
                      class="bx--overflow-menu"
                      data-testid="configuration-property-table-overflow-menu"
                      role="button"
                      tabindex="0"
                    >
                      <svg
                        aria-label="Overflow menu icon"
                        class="bx--overflow-menu__icon"
                        focusable="false"
                        height="16"
                        preserveAspectRatio="xMidYMid meet"
                        role="img"
                        style="will-change: transform;"
                        viewBox="0 0 16 16"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <circle
                          cx="8"
                          cy="3"
                          r="1"
                        />
                        <circle
                          cx="8"
                          cy="8"
                          r="1"
                        />
                        <circle
                          cx="8"
                          cy="13"
                          r="1"
                        />
                        <title>
                          Overflow menu icon
                        </title>
                      </svg>
                    </div>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="bx--pagination"
        >
          <div
            class="bx--pagination__left"
          >
            <label
              class="bx--pagination__text"
              for="bx-pagination-select-1"
              id="bx-pagination-select-1-count-label"
            >
              Items per page:
            </label>
            <div
              class="bx--form-item"
            >
              <div
                class="bx--select bx--select--inline bx--select__item-count"
              >
                <div
                  class="bx--select-input--inline__wrapper"
                >
                  <div
                    class="bx--select-input__wrapper"
                  >
                    <select
                      class="bx--select-input"
                      id="bx-pagination-select-1"
                    >
                      <option
                        class="bx--select-option"
                        value="10"
                      >
                        10
                      </option>
                      <option
                        class="bx--select-option"
                        value="25"
                      >
                        25
                      </option>
                      <option
                        class="bx--select-option"
                        value="50"
                      >
                        50
                      </option>
                    </select>
                    <svg
                      aria-label="open list of options"
                      class="bx--select__arrow"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      style="will-change: transform;"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6l.7-.7L8 9.6l4.3-4.3.7.7z"
                      />
                      <title>
                        open list of options
                      </title>
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="bx--pagination__text"
            >
              1–1 of 1 items
            </span>
          </div>
          <div
            class="bx--pagination__right"
          >
            <div
              class="bx--form-item"
            >
              <div
                class="bx--select bx--select--inline bx--select__page-number"
              >
                <label
                  class="bx--label bx--visually-hidden"
                  for="bx-pagination-select-3"
                >
                  Page number, of 1 pages
                </label>
                <div
                  class="bx--select-input--inline__wrapper"
                >
                  <div
                    class="bx--select-input__wrapper"
                  >
                    <select
                      class="bx--select-input"
                      id="bx-pagination-select-3"
                    >
                      <option
                        class="bx--select-option"
                        value="1"
                      >
                        1
                      </option>
                    </select>
                    <svg
                      aria-label="open list of options"
                      class="bx--select__arrow"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      style="will-change: transform;"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M8 11L3 6l.7-.7L8 9.6l4.3-4.3.7.7z"
                      />
                      <title>
                        open list of options
                      </title>
                    </svg>
                  </div>
                </div>
              </div>
            </div>
            <span
              class="bx--pagination__text"
            >
              of 1 pages
            </span>
            <button
              aria-label="Previous page"
              class="bx--pagination__button bx--pagination__button--backward bx--pagination__button--no-index"
              disabled=""
              type="button"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                height="24"
                preserveAspectRatio="xMidYMid meet"
                style="will-change: transform;"
                viewBox="0 0 32 32"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19 23l-8-7 8-7v14z"
                />
              </svg>
            </button>
            <button
              aria-label="Next page"
              class="bx--pagination__button bx--pagination__button--forward bx--pagination__button--no-index"
              disabled=""
              type="button"
            >
              <svg
                aria-hidden="true"
                focusable="false"
                height="24"
                preserveAspectRatio="xMidYMid meet"
                style="will-change: transform;"
                viewBox="0 0 32 32"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13 9l8 7-8 7V9z"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ReactModalPortal"
  />
</body>
`;
